lib/task_listener.js

```diff
@@ -164,7 +168,8 @@ export default class TaskListener extends EventEmitter {
     if (exceedsThreshold) return;
 
     let claims = await this.taskQueue.claimWork(availableCapacity);
-    claims.forEach(this.runTask.bind(this));
+    let tasksets = await Promise.all(claims.map(this.coalesceClaim.bind(this)));
+    await Promise.all(tasksets.map(this.runTaskset.bind(this)));
```

Comment:
I did see a case where `this.runtime.stats.record` failed with an un-caught error.  But this probably isn't my issue to fix, so I changed the map to a forEach and thus basically left it as I found it.